mypy/messages.py

```diff
@@ -2421,6 +2421,10 @@ def annotation_in_unchecked_function(self, context: Context) -> None:
             code=codes.ANNOTATION_UNCHECKED,
         )
 
+    def type_parameters_should_be_declared(self, undeclared: list[str], context: Context) -> None:
+        names = ", ".join('"' + n + '"' for n in undeclared)
+        self.fail(f"All type parameters should be declared ({names} not declared)", context)
```

Comment:
Should this error message more explicitly refer to "mixing" PEP 695 type variables vs legacy type variables?

I'm sure users would figure it out, but the legacy type variables are "declared" in some sense so it may confuse folks.

(Also I realize fine tuning the error messages can be done later, I just was briefly confused when skimming)

Reply:
I think this is fine for now; I couldn't think of a better wording immediately.